### ISSUE TITLE
Cumulative line chart displays wrong y values

### ISSUE THREAD

**Issue by alexis-pasialis** [novus/nvd3#695]
Has anyone managed to solve the issue described at http://stackoverflow.com/questions/26259954/nvd3-cumulative-line-chart-displays-wrong-y-values ?

Thanks in advance.

**Comment by liquidpele** [comment]
The series (d) should be an object not an array, and by default just make the x an y values as attributes of the object using 'x' and 'y' as the attribute name.

**Comment by alexis-pasialis** [comment]
Has anyone managed to find a solution for this? I have tried liquidpele's suggestion but I still have the same problem. Any help will be greatly appreciated.

**Comment by alexis-pasialis** [comment]
How do I re-open this ticket? I don't want to create a duplicate.

**Comment by liquidpele** [comment]
If you provide a jsfiddle (or similar) displaying the issue using the latest dev branch code then I can re-open this.

**Comment by alexis-pasialis** [comment]
This is it:
http://jsfiddle.net/apasiali/1thf2sbs/

The chart has been converted from a line chart. The max y axis value should be around 140.00 but the cumulative displays 0.40 instead.

Thanks

**Comment by robinfhu** [comment]
Yes, this is very weird.  If you change it to a lineChart, it works correctly.

Cumulative line chart behavior has always been a mystery to me (I didn't develop it).  Maybe it has something to do with the indexify function?  Have you tried debugging it?

**Comment by liquidpele** [comment]
Interesting, it's saving the original x value as x2 on the text element, and converting the data... I'm not sure what exactly it's *supposed* to do either to be honest :p

Re-opening this for discussion/etc I guess.

**Comment by robinfhu** [comment]
@alexis-pasialis  you say this is for a project. Do you know what the Cumulative line chart does?  How does it solve the problem, that a regular line chart can't solve?

**Comment by alexis-pasialis** [comment]
With cumulative charts looking at the data over different time periods can give very different impressions. E.g. if one series has been growing by 10% a year for 3 years, and then by 0% for 2 years, it will immensely outperform a series that has been growing by a steady 2% per year. But if you look over 2 years, the picture looks quite different. Which is why the rescaling is so valuable.

**Comment by yyhy343** [comment]
I am facing the same issue, is there any update on the issue?

**Comment by liquidpele** [comment]
it's on the todo but so is a lot of other stuff... feel free to fix it and send us a pull request if you have time :)

**Comment by jdlubrano** [comment]
I was looking over the cumulative line chart code, and I feel as though the chart's implementation is aimed toward being an [index chart](http://mbostock.github.io/protovis/ex/index-chart.html).  Do you agree?  If so, I have a good handle on how to fix the indexify function and will submit a pull request ASAP.  If the cumulative line chart is supposed to be an index chart, I do think that the model should be renamed accordingly.

I think that a true cumulative line chart is currently more easily achievable by the developer if the underlying data is transformed to be cumulative.  The data can then be added to normal line chart or a linePlusBar chart.  Any thoughts?

**Comment by ghs** [comment]
Looks like this issue is almost 2 years old and still persist. I've spent some hours until surrendered and arrived here.... :(

In my case, any initial data within the graphic is displayed correct, but if there is an update, the data makes no sense just as @vinaymavi's example.
Want to say that not every data set which will be used to the update is generated wrongly.

Example:

`[
[1467158400000, 1],
[1467244800000, 50],
[1467331200000, 50]
]`
though is looking strange because the numbers are 50 and the graph shows 25: 
![img1](https://cloud.githubusercontent.com/assets/94248/16586551/cd1832ce-42c6-11e6-968c-ad957af5f695.png)

But if we have
`[
[1467158400000, 51],
[1467244800000, 50],
[1467331200000, 50]
]`
the result is...
![img2](https://cloud.githubusercontent.com/assets/94248/16586569/f122b770-42c6-11e6-9492-0ce741f45f6a.png)

Any thoughts? I am with @jdlubrano, but have seen that there've been changes around the indexify methods around, so no idea from where could come.

**Comment by liquidpele** [comment]
If anyone figures out a fix please send us a pull request :)

**Comment by NatasaPeic** [comment]
I'm working on the same issue, and I get all values correct on the line chart if my very first value is 0. I can't find the way around it.

**Comment by liquidpele** [comment]
Someone just fixed some stuff in the cumulative line chart recently (https://github.com/novus/nvd3/pull/2014), not sure if that's related but might want to check the latest master branch code to see if this is fixed already.

**Comment by NatasaPeic** [comment]
Thank you @liquidpele. I tried incorporating changes, but I got the same error over and over again -- "currentDomain is not defined". Anyway, cumulative chart wasn't a good pick anyway, so I switched to interactive line chart and it works perfectly. Thanks again for your fast response.
